### PR TITLE
Survivor Quality of Life Update and Tweaks

### PR DESCRIPTION
**Pull request by PalJohnson**
## About The Pull Request

This update seeks to clean up some of the survivor code, fix any bugs from previous updates to survivor and server-related stuff, and tweaks the various load outs of certain items at the request of community feedback

## Why Is It Good For The Game

In short, I've reduced the survivor code by quite a few lines of code to work more efficiently compared to before. In addition, several quality of life changes were implemented due to various overlapping updates not compensating for each other. Cleaner code, changes from community feedback, and neat items thrown is a win in my book.

## What Changed

image

_**General Survivor Quality of Life Changes**_
- Engineering survivor now includes a welder in the satchel due to the removal of the tool belts due to code optimization.
- Allocated space in satchel by moving various pieces of kits to different pouches and belts.
- Special weapon spawns (essentially any weapon other than shotgun spawns) now include an ammo belt. One of the biggest reasons for the survivors ditching their primary weapon was the fact it took too much space to spawn in addition to preventing critical survival gear from spawning. This should alleviate the issue.
- Fixed issue were generic survivors didn't spawn with winter gear. 
- Survivors now spawn with heat absorbant coif when on a cold weather map instead of rebreather.

_**Code Clean Up**_

- Merged the two roughneck survivors into one. It didn't make sense to have two different type of survivors now that we have the random weapon generator online,
- Survival pouches include radios and bayonets reducing the number of lines of code need to fulfill the basic kit survivor needs.
- Survivors no longer spawn with belts. This cleans up the code quite a bit and allows for special weapon spawns to use the belt slot. This also came to the cost of various security survivors who spawned with pistols such as the colonial marshal, security, pmc, and cl survivors.
- Survivors now spawn with an additional magazine of pistol ammo to help compensate for the loss of pistol belts.
- Russian security survivor now no longer spawns with a hunting rifle. They now spawn with weapons like the rest of the survivors using the random weapon spawner.
- Moved tools pouch and survival pouch to spawn on parent survivor. Cleaned up a lot of code that way.

_**Bug Fixes**_

- Corporate Supervisor survivor now spawns with Corporate Liaison loot and Research loot.
- Fixed Researcher Survivor not being able to do surgery despite having identical skills compared to its doctor counterpart.
- Fixed default survivors not being able to spawn with winter gear due to it being unmaintained.
- CL Survivor types now all spawn with CL loot. For some reason, the parent CL did not distribute the loot among all the children. Stingy bastard.

**_Survival Pouches (Human and Synth)_**

- Survival Pouches now include a bayonet and emergency radio. This was done as a result of various loot items all spawning in the satchel. Items such as pistols, pistol ammo, spawned loot (default), special character loot, and spawned weapons meant that players would not receive their items as intended. Given that all survivors spawn with knives and radios I thought it was best to include it in the survivor pouch to help alleviate these issues. This in turn also cleared up various lines of code as each individual survivor had to have 2 lines of code manually adding it to their persona. Updated sprite as well to reflect the change.

**_Weapon Pool_**

- One of the biggest critiques of the current weapon pool was that it heavily emphasized and favored shotgun meta. It had a 75% chance of spawning. Now the CMB shotgun spawns at a rate of 50% with the rest of the weapons being occupied by the standard weapons like MAR40, M16, Hunting rifle, DB, and custom shotgun. When I initially did this left shotguns with a higher probability of spawning due to DB and custom shotguns. To counteract this survivors now can spawn with the FP9000 (SMG) and MP5 (SMG) to help offset the shotgun spawn rate percentage. If survivors want consistent shotgun spawns they will now have to rely on colony armories **_AND SHARE._** The goal is to ultimately remove the CMB shotgun nearly entirely from the spawn pool with the liberator shotgun once more development time can be allocated to it.

**_PMC Survivor_**

- With the loss of the MOD88 belt among other items in the PMC survivor loadout the role lacked unique items specific to the survivor. To compensate for the various loss of items PMC survivors now spawn with weapons from a pool of guns that a PMC might have in the outer rim of the colony. These weapons include the NSG23 (ID locked + No Attachments), MP5, MAR40,  CMB Shotgun, and Custom Shotgun. The PMC survivor deployed to the edges of the galaxy may not be well equipped as their ERT or space station counterparts, however, they are equipped with some of the better weapons the colony has to offer. 20% chance of spawning with either weapon.

**_Prisoner and Gang Leader Survivor_**

- Arguably one of the more underwhelming survivor types second to the colonist they received a minor tweak to their loadout. Both survivors now have a 25% chance of spawning with either an armored vest or helmet. Jailbreaking without appropriate gear is suicide especially when the guards left the station like a bunch of cowards after the xenomorphs attacked.

**_Other Changes_**

- Ammo belts have been created for weapons commonly found on the colony including AP, extended, and drum magazine varrients.
- Doctor survivor's medical loot table has been adjusted to compensate for the loss of medical belts. The now can spawn with a "robust medical first aid kit" which spawns with bic, kelo, trauma, adv brute, adv burn, splints, and health analyzer. Due to the loss of both of the belts, medical survivors have a 25% chance of spawning with surgical first aid kits.
- The chaplain's Holiday suit now has a weapon slot. This should encourage chaplains to maintain uniform while in the thick of praying and fighting.
- Prisoner survivor now has CQC skills like the gang leader survivor at the request of Geeves

## Changelog

:cl: theflyingflail
add: Added New Weapon Pool for PMC Survivors
add: Prisoner survivors now have a chance to spawn with a helmet or weapon
add: FP9000 and MP5 have been added to the survivor spawn pool to help combat shotgun meta while adding variety to weapon loadouts
del: Removed old things
qol: Reduced amount of lines required for survivors to spawn with essential gear
qol: Survival pouches now include radios and knives
qol:  Unique survivor spawn weapons now include ammo belts to help free up space
tweak: PMC survivors now draw from a separate weapons pool dedicated to PMC survivors
tweak: PMC survivors no longer spawn with pmc mask
tweak: Survivors no longer spawn with pistol belts or any belts
tweak: Survivors doctors have updated medical loot spawn due to loss of belts
tweak: Moved the large stack of metal from survivor pouches to the survivor's satchel
tweak: tweaked various survivor skills and load-outs at the request of devs
balance: CMB Shotguns have been reduced to 50% spawn chance as opposed to 75%
fix: Various bugs related to winter gear not spawning and CL loot
fix: Research survivors now can do surgery again
fix: Fixed chaplain survivor being unable to store a weapon in robes
fix: Fixed issue were survivors didn't spawn with winter gear

/:cl:

### CODE REVIEW

**Comment by MusselMain** [comment]
Very nice change, This was needed. The way gear spawned on survivors (especially certain rifles) was incredibly annoying. Now all we need for perfection is for the KT-42 and CMB Autorevovler to get the well deserved buff they've needed, as they both suffer from a fire rate worse then the Desert Eagle, two thirds (KT-42) or half the damage (The CMB does less than a .22 holdout pistol), and yet the KT-42 is somehow a Freelancer, Survivor, and CLF weapon. I can understand if the CMB is meant to be a joke weapon, as it's not forced upon anyone, but the KT-42 is a weapon that either needs to be removed from the sidearm pool, or reworked.

**Comment by PalJohnson** [comment]
Quick Update: I moved the metal from the survivor pouch to spawn in the satchels instead after someone told me that the pouches could be purchased at req. The pouches could hold 350 metal with the pouch expansion. This alleviates the issue.

**Comment by Chainsawicus** [comment]
> _**General Survivor Quality of Life Changes**_
> 
> * The chaplain's Holiday suit now has a weapon slot. This should encourage chaplains to maintain uniform while in the thick of praying and fighting.
> * Prisoner survivor now has CQC skills like the gang leader survivor at the request of Geeves
> * Engineering survivor now includes a welder in the satchel due to the removal of the tool belts due to code optimization.
> * Allocated space in satchel by moving various pieces of kits to different pouches and belts.
> * Special weapon spawns (essentially any weapon other than shotgun spawns) now include an ammo belt. One of the biggest reasons for the survivors ditching their primary weapon was the fact it took too much space to spawn in addition to preventing critical survival gear from spawning. This should alleviate the issue.
> * Fixed issue were generic survivors didn't spawn with winter gear.
> * Survivors now spawn with heat absorbant coif when on a cold weather map instead of rebreather.

With the exception of the last two and the 4th one, these are buffs, not QOL. It's important to note I don't have an issue with any of them, but it should be changed to properly tag them as buffs.

**Comment by Diegoflores31** [comment]
Survivors not having belts will heavily affect doctor survs since they wont spawn with lifesaver belt anymore??

**Comment by PalJohnson** [comment]
> Survivors not having belts will heavily affect doctor survs since they wont spawn with lifesaver belt anymore??

I can make a robust first aid kit that only survivors spawn with to help compensate for potential loss. Much like the survival pouch used to be. Would include basic meds and healing items survivors might use.

**Comment by PalJohnson** [comment]
Some community-based update requests

PMC Survivor no longer spawns with PMC Mask due to face-hugger protection. They are replaced with heat coif which looks the same.
Doctor survivors have adjusted loot tables due to belt changes. They now can spawn with robust first aid kit and now surgical kit boxes to compensate for the loss of medical belts.
image

**Comment by PalJohnson** [comment]
> 



> > _**General Survivor Quality of Life Changes**_
> > 
> > * The chaplain's Holiday suit now has a weapon slot. This should encourage chaplains to maintain uniform while in the thick of praying and fighting.
> > * Prisoner survivor now has CQC skills like the gang leader survivor at the request of Geeves
> > * Engineering survivor now includes a welder in the satchel due to the removal of the tool belts due to code optimization.
> > * Allocated space in satchel by moving various pieces of kits to different pouches and belts.
> > * Special weapon spawns (essentially any weapon other than shotgun spawns) now include an ammo belt. One of the biggest reasons for the survivors ditching their primary weapon was the fact it took too much space to spawn in addition to preventing critical survival gear from spawning. This should alleviate the issue.
> > * Fixed issue were generic survivors didn't spawn with winter gear.
> > * Survivors now spawn with heat absorbant coif when on a cold weather map instead of rebreather.
> 
> With the exception of the last two and the 4th one, these are buffs, not QOL. It's important to note I don't have an issue with any of them, but it should be changed to properly tag them as buffs.

I agree with the Holiday vest issue. Ill changed the description of it on MR. But the rest I see them more as QOL. The removable of belts meant you lost medical gear and engineering equipment for arguably 2 of the strongest survivor type. Things had to be moved around so that their kit could somewhat stay the same. Survival pouches not having metal capacity means you can no longer horde 250 metal in a single pouch.

File: code/modules/gear_presets/_select_equipment.dm [hunk 1]
```diff
@@ -831,104 +821,111 @@ var/list/rebel_rifles = list(
...
-			H.equip_to_slot_or_del(new /obj/item/weapon/melee/twohanded/fireaxe(H), WEAR_R_HAND)
+			H.equip_to_slot_or_del(new /obj/item/weapon/melee/twohanded/fireaxe(H), WEAR_IN_BACK)
 		if(4)
-			H.equip_to_slot_or_del(new /obj/item/tool/weldingtool/largetank(H), WEAR_R_HAND)
+			H.equip_to_slot_or_del(new /obj/item/tool/weldingtool/largetank(H), WEAR_IN_BACK)
 		if(5)
-			H.equip_to_slot_or_del(new /obj/item/storage/firstaid/adv(H.back), WEAR_R_HAND)
+			H.equip_to_slot_or_del(new /obj/item/storage/firstaid/adv(H.back), WEAR_IN_BACK)
 
+/datum/equipment_preset/proc/add_random_prison_survivor_armor(var/mob/living/carbon/human/H) // Random set of armor a prisioner might repurpose in the event of a breakout
```
Comment (Geevies) [inline code/modules/gear_presets/_select_equipment.dm:839]: why is the var here called random_weapon if its adding armor, and why is a bayonet one of the options

File: code/modules/gear_presets/_select_equipment.dm [hunk 2]
```diff
@@ -831,104 +821,111 @@ var/list/rebel_rifles = list(
...
+	var/random_weapon = rand(0,1)
+	switch(random_weapon)
+		if(0)
+			H.equip_to_slot_or_del(new /obj/item/clothing/suit/armor/vest(H.back), WEAR_JACKET)
+		if(1)
+			H.equip_to_slot_or_del(new /obj/item/attachable/bayonet(H), WEAR_IN_BACK)
+			
+
+/datum/equipment_preset/proc/add_random_prison_survivor_helmet(var/mob/living/carbon/human/H) // Random helmet a prisioner might repurpose in the event of a breakout
```
Comment (Geevies) [inline code/modules/gear_presets/_select_equipment.dm:848]: ditto

File: code/modules/gear_presets/survivors.dm [hunk 3]
```diff
@@ -511,18 +475,30 @@
 
 /datum/equipment_preset/survivor/prisoner/load_gear(mob/living/carbon/human/H)
 	H.equip_to_slot_or_del(new /obj/item/clothing/under/color/orange(H), WEAR_BODY)
+	if(SSmapping.configs[GROUND_MAP].environment_traits[MAP_COLD])
+		add_ice_colony_survivor_equipment(H)
 	H.equip_to_slot_or_del(new /obj/item/storage/backpack/satchel(H), WEAR_BACK)
-	H.equip_to_slot_or_del(new /obj/item/attachable/bayonet(H.back), WEAR_IN_BACK)
-	H.equip_to_slot_or_del(new /obj/item/device/radio(H), WEAR_IN_BACK)
+	add_random_prison_survivor_helmet(H)
```
Comment (Geevies) [inline code/modules/gear_presets/survivors.dm:481]: i think a better thing for this proc would be just to remove it and add `if(prob(50)) H.equip_to_slot_or_del(new /obj/item/clothing/suit/armor/vest(H.back), WEAR_JACKET)`

**Review by Geevies — CHANGES_REQUESTED** [review]
besides the things i listed, everything else looks good

**Pull request merged**